src/certificates.c

```diff
@@ -65,13 +66,13 @@ uint16_t
 xtt_server_certificate_length_fromsignaturetype(xtt_server_signature_type type)
 {
     switch (type) {
-        case XTT_SERVER_SIGNATURE_TYPE_ED25519:
+        case XTT_SERVER_SIGNATURE_TYPE_ECDSAP256:
             return sizeof(xtt_identity_type)
                        + sizeof(xtt_certificate_expiry)
                        + sizeof(xtt_certificate_root_id)
-                       + sizeof(xtt_ed25519_pub_key)
-                       + sizeof(xtt_ed25519_signature);
-    }
+                       + sizeof(xtt_ecdsap256_pub_key)
+                       + sizeof(xtt_ecdsap256_signature);
+    }              //returns 169 when using ECDSA_P256
```

Comment:
By the way, in C, you can create scoped variables in a `case` statement by adding brackets:
```
switch (type {
    case XTT_SERVER_SIGNATURE_TYPE_ECDSAP256:
    {
        uint16_t ret = ...
    }
}
```
In general, in C, surrounding some statements in curly brackets introduces a new scope, which means you can declare new variables in that scope and they won't leak out. I like this, because we're making sure our variables only exist where they need to.